test/api/liskApi.js

```diff
@@ -695,4 +426,45 @@ describe('Lisk.api()', () => {
 			});
 		});
 	});
+
+
+	describe('#constructRequestData', () => {
+		const { address } = defaultAddress;
+		const optionsObject = {
+			limit: defaultRequestLimit,
+			offset: defaultRequestOffset,
+		};
+		const expectedObject = {
+			address: '18160565574430594874L',
+			limit: defaultRequestLimit,
+			offset: defaultRequestOffset,
+		};
+		const optionsWithConflictObject = {
+			address: '123L',
+			limit: 4,
+			offset: 5,
+		};
+		const resolvedConflictObject = {
+			address: '123L',
+			limit: defaultRequestLimit,
+			offset: defaultRequestOffset,
+		};
+
+		it('should merge a data object with an options object', () => {
+			const requestData = privateApi.constructRequestData({ address }, optionsObject);
+			(requestData).should.be.eql(expectedObject);
+		});
+
+		it('should recognise when a callback function is passed instead of an options object', () => {
+			const requestData = privateApi.constructRequestData({ address }, () => true);
+			(requestData).should.be.eql({ address });
+		});
+
+		it('should prioritise values from the data object when the data object and options object conflict', () => {
+			const requestData = privateApi.constructRequestData(
+				{ limit: defaultRequestLimit, offset: defaultRequestOffset }, optionsWithConflictObject,
```

Comment:
This is the other way round to what I suggested, but it's ok with me since it tests the conflict.